patches/net/minecraft/server/level/ServerEntity.java.patch

```diff
@@ -12,31 +12,39 @@
                  if (mapitemsaveddata != null) {
                      for (ServerPlayer serverplayer : this.level.players()) {
                          mapitemsaveddata.tickCarriedBy(serverplayer, itemstack);
-@@ -273,22 +_,25 @@
+@@ -273,6 +_,7 @@
      public void removePairing(ServerPlayer p_8535_) {
          this.entity.stopSeenByPlayer(p_8535_);
          p_8535_.connection.send(new ClientboundRemoveEntitiesPacket(this.entity.getId()));
 +        net.neoforged.neoforge.event.EventHooks.onStopEntityTracking(this.entity, p_8535_);
      }
  
      public void addPairing(ServerPlayer p_8542_) {
-         List<Packet<? super ClientGamePacketListener>> list = new ArrayList<>();
--        this.sendPairingData(p_8542_, list::add);
-+        this.sendPairingData(p_8542_, new net.neoforged.neoforge.network.bundle.PacketAndPayloadAcceptor<>(list::add));
+@@ -280,15 +_,17 @@
+         this.sendPairingData(p_8542_, list::add);
          p_8542_.connection.send(new ClientboundBundlePacket(list));
          this.entity.startSeenByPlayer(p_8542_);
 +        net.neoforged.neoforge.event.EventHooks.onStartEntityTracking(this.entity, p_8542_);
      }
  
 -    public void sendPairingData(ServerPlayer p_289562_, Consumer<Packet<ClientGamePacketListener>> p_289563_) {
-+    public void sendPairingData(ServerPlayer p_289562_, net.neoforged.neoforge.network.bundle.PacketAndPayloadAcceptor<net.minecraft.network.protocol.game.ClientGamePacketListener> p_289563_) {
++    public void sendPairingData(ServerPlayer p_289562_, Consumer<Packet<? super ClientGamePacketListener>> p_289563_) {
```

Comment:
What happened here? Was this supposed to be reverted to `Consumer<Packet<ClientGamePacketListener>>` or is it intentionally `? super ...`

Reply:
That's intentional. I managed to simplify this a bit but couldn't fully remove this patch because custom payload packets have a common listener generic.